src/Chord.hs

```diff
@@ -45,7 +45,25 @@ data Note
  | E
  | F
  | G
- deriving (Show, Read, Enum, Bounded)
+ deriving (Show, Read)
+
+instance Enum Note where
+  toEnum 0 = A
+  toEnum 1 = B
+  toEnum 2 = C
+  toEnum 3 = D
+  toEnum 4 = E
+  toEnum 5 = F
+  toEnum 6 = G
+  toEnum n = toEnum $ n `mod` 7
+
+  fromEnum A = 0
+  fromEnum B = 1
+  fromEnum C = 2
+  fromEnum D = 3
+  fromEnum E = 4
+  fromEnum F = 5
+  fromEnum G = 6
```

Comment:
Like we discussed, deriving `Enum` and `Bounded` wasn't the move here; we can give a custom implementation of `Enum` so that `succ` and `pred` wrap around appropriately.